packages/components/src/DateTimePickers/Date/Manager/Manager.component.js

```diff
@@ -11,32 +11,35 @@ import {
 } from '../date-extraction';
 
 function ContextualManager(props) {
-	// eslint-disable-next-line no-use-before-define
+	function getDateOptions() {
+		return {
+			dateFormat: props.dateFormat,
+			useUTC: props.useUTC,
+			required: props.required,
+			timezone: props.timezone,
+		};
+	}
+
 	const initialState = extractDate(props.value, getDateOptions());
 	const [state, setState] = useState(initialState);
+
 	useEffect(() => {
-		// eslint-disable-next-line no-use-before-define
-		const newState = extractDate(props.value, getDateOptions());
-		setState(newState);
+		if (props.value !== state.date) {
+			const newState = extractDate(props.value, getDateOptions());
+			setState(newState);
+		}
 	}, [props.value]);
 
 	useEffect(() => {
 		checkSupportedDateFormat(props.dateFormat);
 	}, [props.dateFormat]);
+
 	useEffect(() => {
 		if (props.timezone) {
 			checkSupportedTimezone(props.timezone);
 		}
 	}, [props.timezone]);
 
-	function getDateOptions() {
-		return {
-			dateFormat: props.dateFormat,
-			useUTC: props.useUTC,
-			required: props.required,
-			timezone: props.timezone,
-		};
-	}
```

Comment:
move this function definition to the top to get rid of eslint `no-use-before-define` warning